backend/metadata_writer/src/metadata_writer.py

```diff
@@ -393,6 +397,12 @@ def is_kfp_v2_pod(pod) -> bool:
 
                 pods_with_written_metadata[obj.metadata.name] = None
 
-        except Exception as e:
-            import traceback
-            print(traceback.format_exc())
+        # Server side timeout, continue watching.
+
+    except urllib3.exceptions.ReadTimeoutError as e:
+        # Client side timeout, continue watching.
+        pass
+
+    except Exception as e:
```

Comment:
I know we were already catching all these errors before, but I am struggling to see why catching `Exception` won't get us sometimes stuck where we never actually crash.

Especially because you proposed above that we have this try around the for loop, meaning any iteration errors will endlessly retry on a `pod_stream` that may never work?

Reply:
I wanted to keep the amount of changes small originally.

But in my understanding removing this catch-all except clause improves the code. The unhalded exceptions will still be printed to the console and they will not be hidden from a user anymore. A user will see that restart counter increases and will be able to get the logs by a commnad like `kubectl -n kubeflow logs --previous metadata-writer-6d5b8456-78265`. Kubernetes will restart `metadata-writer` pod so it will continue handling events if it is possible. 

So I made a commit that removes this.